azure/scope/identity.go

```diff
@@ -351,3 +363,112 @@ func IsClusterNamespaceAllowed(ctx context.Context, k8sClient client.Client, all
 
 	return false
 }
+
+func reconcileASOSecret(ctx context.Context, azureClusterIdentity *infrav1.AzureClusterIdentity, kubeClient client.Client, subscriptionID string) error {
+	asoSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      identity.GetASOSecretName(subscriptionID, azureClusterIdentity.Name),
+			Namespace: azureClusterIdentity.Namespace,
+			Labels: map[string]string{
+				infrav1.ASOSecretLabel: azureClusterIdentity.Name,
+			},
+		},
+		Data: map[string][]byte{
+			"AZURE_SUBSCRIPTION_ID": []byte(subscriptionID),
+			"AZURE_TENANT_ID":       []byte(azureClusterIdentity.Spec.TenantID),
+			"AZURE_CLIENT_ID":       []byte(azureClusterIdentity.Spec.ClientID)},
+	}
+
+	apiVersion, kind := infrav1.GroupVersion.WithKind(azureClusterIdentity.Kind).ToAPIVersionAndKind()
+	owner := metav1.OwnerReference{
+		APIVersion: apiVersion,
+		Kind:       kind,
+		Name:       azureClusterIdentity.GetName(),
+		UID:        azureClusterIdentity.GetUID(),
+	}
+
+	// Fetch identity secret, if it exists
+	key := types.NamespacedName{
+		Namespace: azureClusterIdentity.Spec.ClientSecret.Namespace,
+		Name:      azureClusterIdentity.Spec.ClientSecret.Name,
+	}
+	identitySecret := &corev1.Secret{}
+	err := kubeClient.Get(ctx, key, identitySecret)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return errors.Errorf("failed to fetch identity secret %s as defined in AzureClusterIdentity %q/%q: %v", identitySecret.Name, azureClusterIdentity.Namespace, azureClusterIdentity.Name, err)
+	}
+
+	// TODO add support for workload identity
```

Comment:
@adriananeci Could you please either unlink #3526 from this PR so that issue doesn't automatically close or open a new issue for Workload ID + ASO to make sure this is tracked?

Reply:
Sorry for the back and forth @adriananeci but I synced with @dtzar and could you please re-link #3526 to this PR so it gets closed once this is merged? I've opened a separate issue for just workload ID (#3732).